v1/src/main/java/com/google/cloud/teleport/spanner/AvroUtil.java

```diff
@@ -68,4 +70,33 @@ public static Schema unpackNullable(Schema schema) {
     }
     return null;
   }
+
+  /**
+   * For named schema, the database object name could have dot(.). When building a avro schema
+   * object, a name with dot(.) is special. See org.apache.avro.Schema.Name#name for more
+   * detail. So we use avro property to store the database object name in newer version, this
+   * function thus provides a way to get name in a compatible way.
+   *
+   * @return spanner's database object name.
+   */
+  public static String getSpannerObjectName(Schema schema) {
+    if (schema.getProp(SPANNER_NAME) != null && !schema.getProp(SPANNER_NAME).isEmpty()) {
+      return schema.getProp(SPANNER_NAME);
+    }
+    return schema.getName();
+  }
+
+  /**
+   * For a fully qualified database object name, we can not use it directly as avro schema name.
+   * Transform the name by replacing the dot(.) using underscore(_) to avoid conflicts with avro.
```

Comment:
Do we allow underscore (_) character in database names? If it is allowed, this logic won't work.

Reply:
Spanner allows underscore in table names, column names etc. This logic will break if there are underscores in the spannerName.

Reply:
It should be fine. The Avro name is just to make sure it is unique in Avro. For example, if we have sch1.foo and sch2.foo, I need a way to make sure the name is uniquness, that is why I prefix the foo using the schema name.

My testing db has a lot of database objects columns used underscore.